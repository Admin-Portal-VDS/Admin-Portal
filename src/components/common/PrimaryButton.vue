<template>
  <div class="primary-button">Primary Button</div>
</template>

<script>
export default {
  name: 'PrimaryButton',
};
</script>

<style scoped>
.primary-button {
  border: 1.3px solid;
  border-radius: 6px;
  display: inline-block;
  font-size: 1rem;
  font-weight: 590;
  font-style: normal;
  text-align: center;
  padding: 8px 12px;
  min-height: 15px;
  min-width: 60px;
  color: #000;
  appearance: none;
  margin-left: 500px;
}
</style>
