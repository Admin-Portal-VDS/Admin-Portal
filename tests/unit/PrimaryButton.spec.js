import { mount } from '@vue/test-utils';
import PrimaryButton from '@/components/common/PrimaryButton.vue';

describe('PrimaryButton.vue', () => {
  it('renders button name correctly', () => {
    const buttonName = 'PrimaryButton';
    const wrapper = mount(PrimaryButton, {
      propsData: {
        buttonName,
      },
    });
    const title = wrapper.find('button');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe(buttonName);
  });

  it('accepts props and passes them correctly', () => {
    const buttonName = 'Test Button';
    const wrapper = mount(PrimaryButton, {
      propsData: { buttonName },
    });
    expect(wrapper.props('buttonName')).toBe(buttonName);
  });

  it('throws an error if required prop is missing', () => {
    console.error = jest.fn();
    expect(() => {
      mount(PrimaryButton);
    }).toThrow('Label prop is required');
  });

  it('matches snapshot', () => {
    const wrapper = mount(PrimaryButton, {
      propsData: {
        buttonName: 'Snapshot Test', // Correct prop name
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
