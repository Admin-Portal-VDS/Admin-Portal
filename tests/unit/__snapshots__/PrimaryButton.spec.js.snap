// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton.vue matches snapshot 1`] = `<div><button class="simple-button">Snapshot Test</button></div>`;
